lib/extensions/ar_virtual.rb

```diff
@@ -66,43 +66,99 @@ module ClassMethods
+      unless (to = options[:to])
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
-      delegate(*methods, options)
+
+      to = to.to_s
+      if to.include?(".") && methods.size > 1
+        raise ArgumentError, 'Delegation only supports specifying a method name when defining a single virtual method'
+      end
+
+      location = caller_locations(1, 1).first
+      file, line = location.path, location.lineno
+
+      allow_nil = options[:allow_nil]
+      default = options[:default]
 
       # put method entry per method name.
       # This better supports reloading of the class and changing the definitions
       methods.each do |method|
-        method_prefix = virtual_delegate_name_prefix(options[:prefix], options[:to])
+        method_prefix = virtual_delegate_name_prefix(options[:prefix], to)
         method_name = "#{method_prefix}#{method}"
+        if to.include?(".") # to => "target.method"
+          to, method = to.split(".")
+          options[:to] = to
+        end
+
+        define_delegate(method_name, method,
+                        :to        => to,
+                        :allow_nil => allow_nil,
+                        :file      => file,
+                        :line      => line,
+                        :default   => default)
 
         self.virtual_delegates_to_define =
-          virtual_delegates_to_define.merge(method_name => [method, options])
+          virtual_delegates_to_define.merge(method_name => [to, method])
       end
     end
 
     private
 
-    # @option :methods :to
-    # @option :methods :prefix
-    def define_virtual_delegate(methods, options)
-      unless (to = options[:to]) && (to_ref = reflection_with_virtual(to.to_s))
+    def define_virtual_delegate(method_name, to, col)
+      unless (to_ref = reflection_with_virtual(to.to_s))
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
 
-      prefix = options[:prefix]
-      method_prefix = virtual_delegate_name_prefix(prefix, to)
-
       to_model = to_ref.klass
-      methods.each do |col|
-        col = col.to_s
-        type = to_model.type_for_attribute(col)
-        raise "unknown attribute #{to_model.name}##{col} referenced in #{name}" unless type
-        arel = virtual_delegate_arel(col, to, to_model, to_ref)
-        define_virtual_attribute "#{method_prefix}#{col}", type, :uses => to, :arel => arel
+      col = col.to_s
+      type = to_model.type_for_attribute(col)
+      raise "unknown attribute #{to_model.name}##{col} referenced in #{name}" unless type
+      arel = virtual_delegate_arel(col, to, to_model, to_ref)
+      define_virtual_attribute method_name, type, :uses => to, :arel => arel
+    end
+
+    # see activesupport module/delegation.rb
```

Comment:
Is this a copy of delegation?  If so, my gut is telling me to find another way.  What did you change that this needs to be included here?